### PR TITLE
Update citizen

### PR DESCRIPTION
**Pull request by MaksimHamko**
## Description
You can update a citizen name by swiping to the right inside the list of citizens for that given organisation

## Type of PR
- Feature

## Related Issues/Tickets
https://github.com/aau-giraf/weekplan/issues/86

Please link the related issues/tickets here.

## Checklist

- [x] Completed the user story described by the issue (if applicable)
- [x] Code follows the code guidelines
- [x] Tests have been added/updated (if applicable)
- [x] Documentation has been updated (if applicable)

### CODE REVIEW

**Review by JEEEEENS21 — APPROVED** [review]

File: app/auth/profile/organisation/citizens/[index].tsx [hunk 1]
```diff
@@ -20,12 +38,32 @@ const ViewCitizen = () => {
 
   const filteredData = useSearch(data?.citizens || [], searchQuery, citizenSearchFn);
 
+  const [selectedCitizenId, setSelectedCitizenId] = useState<string | number | null>(null);
+  const [firstName, setFirstName] = useState("");
+  const [lastName, setLastName] = useState("");
```
Comment (MertzAndreas) [inline app/auth/profile/organisation/citizens/[index].tsx:43]: Ligner noget som bør være et enkelt objekt

File: app/auth/profile/organisation/citizens/[index].tsx [hunk 2]
```diff
@@ -20,12 +38,32 @@ const ViewCitizen = () => {
...
   const handleDelete = async (id: number) => {
     await deleteCitizen.mutateAsync(id).catch((error) => {
       addToast({ message: error.message, type: "error" });
     });
   };
 
+  const openBottomSheet = (citizen: Citizen) => {
+    setSelectedCitizenId(citizen.id);
+    setFirstName(citizen.firstName);
+    setLastName(citizen.lastName);
```
Comment (MertzAndreas) [inline app/auth/profile/organisation/citizens/[index].tsx:55]: Vil også se pænere ud som et objekt

File: app/auth/profile/organisation/citizens/[index].tsx
Comment (MertzAndreas) [inline app/auth/profile/organisation/citizens/[index].tsx:1]: Check data go display loading omens

**Review by MertzAndreas — DISMISSED** [review]

File: hooks/useOrganisation.ts [hunk 3]
```diff
@@ -113,6 +114,37 @@ const useOrganisation = (orgId: number) => {
...
+      console.log("Previous data:", previousOrg);
+
+      await queryClient.cancelQueries({ queryKey });
+
+      queryClient.setQueryData<OrgDTO>(queryKey, (oldData) => {
+        if (oldData) {
+          const updatedCitizens = oldData.citizens.map((citizen) =>
+            citizen.id === newCitizen.id ? newCitizen : citizen
+          );
+          console.log("Updated citizens:", updatedCitizens); // Log updated citizens list
```
Comment (DanielRBech) [inline hooks/useOrganisation.ts:132]: NO CONSOLE LOG

File: hooks/useOrganisation.ts [hunk 4]
```diff
@@ -113,6 +114,37 @@ const useOrganisation = (orgId: number) => {
     },
   });
 
+  const updateCitizen = useMutation<void, Error, CitizenDTO>({
+    mutationFn: (citizen) => updateCitizenRequest(Number(citizen.id), citizen.firstName, citizen.lastName),
+    onMutate: async (newCitizen) => {
+      newCitizen.id = Number(newCitizen.id);
+
+      const previousOrg = queryClient.getQueryData<OrgDTO>(queryKey);
+      console.log("Previous data:", previousOrg);
```
Comment (DanielRBech) [inline hooks/useOrganisation.ts:123]: NO CONSOLE LOG

**Review by DanielRBech — APPROVED** [review]

**Review by JEEEEENS21 — APPROVED** [review]

**Review by MertzAndreas — CHANGES_REQUESTED** [review]
Mine first

**Pull request merged**